src/track/track.cpp

```diff
@@ -162,7 +162,7 @@ void Track::setTrackMetadata(
         setBpm(actualBpm.getValue());
     }
 
-    if (!newKey.isEmpty()) {
+    if (!newKey.isEmpty() && newKey != mixxx::track::io::key::INVALID) {
```

Comment:
This line fails to compile
```
src/track/track.cpp:165:37: error: no match for ‘operator!=’ (operand types are ‘const QString’ and ‘mixxx::track::io::key::ChromaticKey’)
     if (!newKey.isEmpty() && newKey != mixxx::track::io::key::INVALID) {
                                     ^
```

Reply:
The condition is hard to reason about. Is it possible to explicitly name it and encapsulate it in a function that takes a key and returns `bool`?

Reply:
This line compiles for me.. different compilers, I guess? 

Uwe, would it make more sense to use `Keys::isValid()` for the second part? newKey is not a Keys object, but should it be? If not, where would be the best place to write such a function?

Reply:
How about `static bool KeyUtils::isValid(mixxx::track::io::key::ChromaticKey)`? `Keys` is not appropriate for this purpose.

Reply:
OK, I can do that, but I also just noticed that there is already `mixxx::track::io::key::ChromaticKey_IsValid`, but I can't figure out how to use it. That `newKey` comes from Track::getKey(), which looks like it should return a ChromaticKey, but when I try to use that function with

```{diff}
-    const auto newKey = trackMetadata.getTrackInfo().getKey();
+    const mixxx::track::io::key::ChromaticKey newKey = trackMetadata.getTrackInfo().getKey();
```

at the top of `setTrackMetadata`, I get `src/track/track.cpp:129:92: error: cannot convert ‘const QString’ to ‘const mixxx::track::io::key::ChromaticKey’ in initialization`